### PR TITLE
Aut 2490/make lambda invoker more generic

### PR DESCRIPTION
**Pull request by BeckaL**
## What

The Account Interventions Handler has been modified to invoke the TICF Lambda dependent on the value of a feature switch.

## How to review

1. Code Review
2. Deploy to sandpit with `./deploy-sandpit.sh -a`

## Checklist

- [x] Impact on orch and auth mutual dependencies has been checked.

### CODE REVIEW

File: frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/lambda/AccountInterventionsHandlerTest.java [hunk 1]
```diff
@@ -322,28 +347,50 @@ void shouldReturn200ForSuccessfulRequestAndSubmitAppropriateAuditEvents(
...
+                        TEST_CLIENT_SESSION_ID,
+                        TEST_SESSION_ID,
+                        TEST_INTERNAL_SUBJECT_ID,
+                        EMAIL,
+                        IP_ADDRESS,
+                        AuditService.UNKNOWN,
+                        DI_PERSISTENT_SESSION_ID,
+                        new AuditService.RestrictedSection(Optional.of(ENCODED_DEVICE_DETAILS)));
+
+        // When
```
Comment (BeckaL) [inline frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/lambda/AccountInterventionsHandlerTest.java]: can we remove this comment?

File: ci/terraform/oidc/authdev1.tfvars [hunk 2]
```diff
@@ -14,6 +14,7 @@ account_intervention_service_call_enabled   = true
 account_intervention_service_action_enabled = true
 account_intervention_service_abort_on_error = true
 send_storage_token_to_ipv_enabled           = true
+call_ticf_cri                               = false
```
Comment (BeckaL) [inline ci/terraform/oidc/authdev1.tfvars:17]: Do we want to set this to true? (I'd argue for yes to make all dev envs consistent)

File: shared/src/main/java/uk/gov/di/authentication/shared/services/ConfigurationService.java [hunk 3]
```diff
@@ -187,6 +187,10 @@ public String getTicfCRILambdaName() {
         return System.getenv().getOrDefault("TICF_CRI_LAMBDA_NAME", "");
     }
 
+    public boolean getInvokeTicfCRILambda() {
```
Comment (BeckaL) [inline shared/src/main/java/uk/gov/di/authentication/shared/services/ConfigurationService.java:190]: How about `isInvokeTicfCRILambdaEnabled`?

**Review by BeckaL — COMMENTED** [review]
Generally LGTM, a few minor comments plus some sonar stuff that's reared it's head.

For future ref, I'd have found this a bit easier to review if some of the changes were in their own commits - e.g. the first commit is quite beefy and contains some logically separate stuff. (Originally I was planning on raising a separate first PR just for the changes to the lambda invoker as a preparatory step, but e.g. that would have at least been useful as a separate commit).

**Review by BeckaL — COMMENTED** [review]
All looks good to me, thanks for this :-)

**Review by andrew-moores — APPROVED** [review]

**Pull request merged**